### PR TITLE
Convolution + Cross-Correlation + Autocorrelation

### PR DESCRIPTION
**Pull request by rprechelt**
Hey again! I've made some developments to Surge that I think might be beneficial to the framework.

1. Implemented convolution with `vDSP_conv()` for float and doubles
2. Implemented cross-correlation with `vDSP_conv()` for float and doubles
3. Implemented auto-correlation with `vDSP_conv()` for float and doubles
4. All functions have been tested against the corresponding MATLAB implementation and I implemented a test suite for all 6 functions (using test data from MATLAB)
5. Implemented an `XCTestCase` for checking array equality with accuracy for float and doubles
6. Adjusted README.md to show inclusion of `sub`, `conv` and `xcorr` in the framework

Let me know if you would like me to make any changes to the code before you accept this PR!

### CODE REVIEW

File: Source/Convolution.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Accelerate
+
+// MARK: Convolution
+
+// Convolution of a signal [x], with a kernel [k]. The signal must be at least as long as the kernel.
+public func conv(x: [Float], _ k: [Float]) -> [Float] {
+    assert(x.count >= k.count, "Input vector [x] must have at least as many elements as the kernel,  [k]")
```
Comment (jasonhotsauce) [inline Source/Convolution.swift]: Maybe `precondition` might be better than `assert`?

Comment (rprechelt) [inline Source/Convolution.swift]: You're right! I'll change those to preconditions this evening.

Comment (rprechelt) [inline Source/Convolution.swift]: Changed!

**Comment by rfdickerson** [comment]
Thanks Remy!

**Pull request merged**